File: src/main/java/play/pluv/login/domain/SocialLoginId.java
package play.pluv.login.domain;

import static jakarta.persistence.CascadeType.PERSIST;
import static jakarta.persistence.GenerationType.IDENTITY;
import static lombok.AccessLevel.PROTECTED;

import jakarta.persistence.Embedded;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.validation.constraints.NotNull;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import play.pluv.member.domain.Member;
import play.pluv.oauth.domain.OAuthMemberInfo;

@Entity
@NoArgsConstructor(access = PROTECTED)
public class SocialLoginId {

  @Id
  @GeneratedValue(strategy = IDENTITY)
  private Long id;
  @ManyToOne(cascade = PERSIST)
  @Getter
  @NotNull
  private Member member;
  @NotNull
  @Embedded
  private OAuthMemberInfo oauthMemberInfo;

  @Builder
  public SocialLoginId(final Member member, final OAuthMemberInfo oauthMemberInfo) {
    this.member = member;
    this.oauthMemberInfo = oauthMemberInfo;
  }
}

File: src/main/kotlin/play/pluv/login/domain/SocialLoginId.kt
package play.pluv.login.domain

import jakarta.persistence.*
import play.pluv.member.domain.Member
import play.pluv.oauth.domain.OAuthMemberInfo
import play.pluv.playlist.domain.MusicStreaming

@Entity
class SocialLoginId(
    member: Member,
    oauthMemberInfo: OAuthMemberInfo
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
        private set

    @ManyToOne(fetch = FetchType.LAZY, cascade = [CascadeType.PERSIST])
    @JoinColumn(name = "member_id", nullable = false)
    var member: Member = member
        private set

    @Embedded
    var oauthMemberInfo: OAuthMemberInfo = oauthMemberInfo
        private set

    val source: MusicStreaming
        get() = oauthMemberInfo.source
}
